### PR TITLE
Add health and ready check

### PR DESCRIPTION
**Pull request by grischperl**
**Description**

Changes proposed in this pull request:

- Add health check
- Add liveness check
- Update the Readme

**Related issue(s)**

#9769

### CODE REVIEW

File: components/eventing-controller/cmd/eventing-controller-nats/main.go [hunk 1]
```diff
@@ -79,6 +83,15 @@ func main() {
 		os.Exit(1)
 	}
 
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
```
Comment (montaro) [inline components/eventing-controller/cmd/eventing-controller-nats/main.go:86]: I would change the hardcoded string literal to a constant

File: components/eventing-controller/cmd/eventing-controller-nats/main.go [hunk 2]
```diff
@@ -79,6 +83,15 @@ func main() {
...
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
```
Comment (montaro) [inline components/eventing-controller/cmd/eventing-controller-nats/main.go:90]: I would change the hardcoded string literal to a constant

File: components/eventing-controller/cmd/eventing-controller-nats/main.go (hunk 1 above)
Comment (grischperl) [reply components/eventing-controller/cmd/eventing-controller-nats/main.go]: Done

File: components/eventing-controller/cmd/eventing-controller-nats/main.go (hunk 2 above)
Comment (grischperl) [reply components/eventing-controller/cmd/eventing-controller-nats/main.go]: Done

**Review by alexandra-simeonova — DISMISSED** [review]

**Review by montaro — DISMISSED** [review]

**Review by sayanh — CHANGES_REQUESTED** [review]
General comments:
- The new flags need to be added in the readme
- The chart changes need to be done with image bumps

File: components/eventing-controller/cmd/eventing-controller/main.go [hunk 3]
```diff
@@ -33,9 +34,15 @@ func main() {
 	setupLog := ctrl.Log.WithName("setup")
 
 	var metricsAddr string
+	var probeAddr string
+	var readyEndpoint string
+	var healthEndpoint string
 	var resyncPeriod time.Duration
 	var enableDebugLogs bool
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
```
Comment (sayanh) [inline components/eventing-controller/cmd/eventing-controller/main.go:43]: (nit): Although the below description is copied from the controller-runtime, it seems complete yet concise to me. 
`is the TCP address that the controller should bind to for serving health probes`

**Comment by Halamix2** [comment]
Recent changes in projwjob definitions trigerred some unnecessary jobs in "Expected" state, `/retest` fixes that

File: components/eventing-controller/cmd/eventing-controller/main.go (hunk 3 above)
Comment (grischperl) [reply components/eventing-controller/cmd/eventing-controller/main.go]: Done, and matched the metrics-address description as well

File: resources/eventing/charts/controller/values.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ image:
   # name is the name of the container image for the eventing-controller
   name: "eventing-controller"
   # tag is the container tag of the eventing-controller image
-  tag: "81e3637e"
+  tag: "PR-10674"
```
Comment (sayanh) [inline resources/eventing/charts/controller/values.yaml:18]: Can you please use the new liveness and readiness endpoints in the controller deployments. E.g. [nats-controller](https://github.com/grischperl/kyma/blob/define-readiness-liveness-for-eventing-controller-9769/resources/eventing/charts/nats-controller/templates/deployment.yaml#L2)

Comment (sayanh) [reply resources/eventing/charts/controller/values.yaml:18]: Turns out [this change](https://github.com/kyma-project/kyma/pull/11050/files#diff-236cf4a612aab7ea5de05440a447d5f33dee7a4d17a008749378273c7b482c67R39) will go in first hence we gotta clean it up with the new endpoints.

File: resources/eventing/charts/controller/templates/deployment.yaml [hunk 5]
```diff
@@ -67,6 +67,26 @@ spec:
...
+              port: {{ .Values.healthProbe.port }}
+              scheme: HTTP
+            initialDelaySeconds: 5
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 2
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: {{ .Values.healthProbe.readyness.service.path }}
```
Comment (sayanh) [inline resources/eventing/charts/controller/templates/deployment.yaml:83]: typo: `readiness`

File: resources/eventing/charts/nats-controller/templates/deployment.yaml [hunk 6]
```diff
@@ -33,17 +33,26 @@ spec:
...
             successThreshold: 1
-            timeoutSeconds: 3
+            timeoutSeconds: 2
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: {{ .Values.healthProbe.readyness.service.path }}
+              port: {{ .Values.healthProbe.port }}
+              scheme: HTTP
+            initialDelaySeconds: 10
```
Comment (sayanh) [inline resources/eventing/charts/nats-controller/templates/deployment.yaml:52]: Can we please template these fields?

File: resources/eventing/charts/nats-controller/templates/deployment.yaml [hunk 7]
```diff
@@ -33,17 +33,26 @@ spec:
...
-            initialDelaySeconds: 60
+            initialDelaySeconds: 5
             periodSeconds: 10
             successThreshold: 1
-            timeoutSeconds: 3
+            timeoutSeconds: 2
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: {{ .Values.healthProbe.readyness.service.path }}
```
Comment (sayanh) [inline resources/eventing/charts/nats-controller/templates/deployment.yaml:49]: typo: `readiness`

File: resources/eventing/charts/nats-controller/templates/deployment.yaml (hunk 6 above)
Comment (grischperl) [reply resources/eventing/charts/nats-controller/templates/deployment.yaml]: done

File: resources/eventing/charts/nats-controller/templates/deployment.yaml (hunk 7 above)
Comment (grischperl) [reply resources/eventing/charts/nats-controller/templates/deployment.yaml]: done

**Review by sayanh — APPROVED** [review]

**Comment by grischperl** [comment]
\retest

**Review by sayanh — APPROVED** [review]

File: components/eventing-controller/README.md [hunk 8]
```diff
@@ -67,12 +67,15 @@ This section explains how to use the Eventing Controller. It expects the followi
...
 
-| Flag                  | Description                                               | Default Value | Backend |
-| --------------------- | --------------------------------------------------------- | ------------- | ------- |
-| `metrics-addr`        | The address the metric endpoint binds to.                 | :8080         | both    |
-| `reconcile-period`    | The period between triggering of reconciling calls (BEB). | 10 minutes    | BEB     |
-| `max-reconnects`      | The maximum number of reconnection attempts (NATS).       | 10            | NATS    |
-| `reconnect-wait`      | Wait time between reconnection attempts (NATS).           | 1 second      | NATS    |
+| Flag                     | Description                                               | Default Value | Backend |
+| ------------------------ | --------------------------------------------------------- | ------------- | ------- |
+| `metrics-addr`           | The TCP address that the controller binds to for serving prometheus metrics.  | :8080 | Both  |
```
Comment (alexandra-simeonova) [inline components/eventing-controller/README.md:72]:
```suggestion
| `metrics-addr`           | The TCP address that the controller binds to for serving Prometheus metrics.  | `:8080` | Both  |
```

File: components/eventing-controller/README.md [hunk 9]
```diff
@@ -67,12 +67,15 @@ This section explains how to use the Eventing Controller. It expects the followi
...
+| `health-probe-bind-addr` | The TCP address that the controller binds to for serving health probes.       | :8080 | Both  |
+| `ready-check-endpoint`   | The endpoint of the readiness probe.                      | readyz        | Both    |
+| `health-check-endpoint`  | The endpoint of the health probe.                         | healthz       | Both    |
```
Comment (alexandra-simeonova) [inline components/eventing-controller/README.md:75]:
```suggestion
| `health-probe-bind-addr` | The TCP address that the controller binds to for serving health probes.       | `:8080` | Both  |
| `ready-check-endpoint`   | The endpoint of the readiness probe.                      | `readyz`        | Both    |
| `health-check-endpoint`  | The endpoint of the health probe.                         | `healthz`       | Both    |
```

**Review by alexandra-simeonova — APPROVED** [review]

**Review by sayanh — APPROVED** [review]

**Pull request merged**